framework/src/materials/MaterialData.C

```diff
@@ -64,10 +64,16 @@ MaterialData::swap(const Elem & elem, unsigned int side /* = 0*/)
 }
 
 void
-MaterialData::reinit(const std::vector<std::shared_ptr<Material>> & mats)
+MaterialData::reinit(const std::vector<std::shared_ptr<Material>> & mats,
+                     bool prevent_update_interface_materials)
 {
   for (const auto & mat : mats)
-    mat->computeProperties();
+  {
+    if (prevent_update_interface_materials && mat->isInterfaceMaterial())
```

Comment:
@lindsayad thanks for reviewing this PR. A clean way to avoid these checks would be to distinguish between boundary and interface materials,e.g. adding  `_interface_material_data`.
Then one can write a new `reinitMaterialsInterface`. I'm not sure about the overhead of this.

Reply:
I do think that would be a solution, and I think it's reasonable. However, I'm not sure @andrsd would like it. The current design is that boundary and face materials both use the same `MaterialData` object (see below). If one was to add `_interface_material_data`, I think you could argue that we should also split `_bnd_material_data` out into `_bnd_material_data` and `_internal_face_material_data`, and I'm just not sure if that's what we want. I believe @andrsd gets back next Monday, and I hope we can resolve it then. Hopefully you can wait until then?
```
std::shared_ptr<MaterialData>
FEProblemBase::getMaterialData(Moose::MaterialDataType type, THREAD_ID tid)
{
  std::shared_ptr<MaterialData> output;
  switch (type)
  {
    case Moose::BLOCK_MATERIAL_DATA:
      output = _material_data[tid];
      break;
    case Moose::NEIGHBOR_MATERIAL_DATA:
      output = _neighbor_material_data[tid];
      break;
    case Moose::BOUNDARY_MATERIAL_DATA:
    case Moose::FACE_MATERIAL_DATA:
      output = _bnd_material_data[tid];
      break;
  }
  return output;
}
```

Reply:
@lindsayad yes I can wait no problem. Now everything is working fine and I can  work on a branch of this PR for doing other stuff. As far as I'm concerned now it's just a matter of making the code cleaner. Also, I want  this to be done as correct as possible as I would like to avoid debugging dependency problems in the future :). It has been a long journey since the beginning but now I can see the light at the end of tunnel and start doing interesting calculations with MOOSE :)